### PR TITLE
JU-566 Java 21 Upgrade

### PR DESCRIPTION
**Pull request by NeilShadrach**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,10 +50,21 @@ sonar:
 sonar-pr-analysis:
 	mvn sonar:sonar -P sonar-pr-analysis
 
-.PHONY: docker-build
-docker-build:
-	docker build -t $(artifact_name):$(version) .
+.PHONY: security-check
+security-check:
+	mvn org.owasp:dependency-check-maven:update-only
+	mvn org.owasp:dependency-check-maven:check -DfailBuildOnCVSS=4 -DassemblyAnalyzerEnabled=false
```
[the remark this answers is not in the document]
Comment (NeilShadrach) [reply Makefile]: Updated

File: src/main/java/uk/gov/companieshouse/payments/admin/web/WebSecurity.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurity {
+
+    @Bean
+    public SecurityFilterChain healthcheckSecurityFilterChain(HttpSecurity http) throws Exception {
+
+        http.securityMatcher("/admin/payments/healthcheck")
+            .csrf(AbstractHttpConfigurer::disable);
```
[the remark this answers is not in the document]
Comment (NeilShadrach) [reply src/main/java/uk/gov/companieshouse/payments/admin/web/WebSecurity.java]: Removed

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by bcullerton — APPROVED** [review]

**Pull request merged**